src/util.jl

```diff
@@ -0,0 +1,179 @@
+    end
+
+    if !(joinpath(brew_prefix,"lib") in DL_LOAD_PATH)
+        push!(DL_LOAD_PATH, joinpath(brew_prefix, "lib") )
+    end
+
+    # We need to set our own, private, cache directory so that we don't conflict with
+    # user-maintained Homebrew installations, and multiple users can use it at once
+    ENV["HOMEBREW_CACHE"] = joinpath(ENV["HOME"],"Library/Caches/Homebrew.jl/")
+
+    # If we have `git` installed from Homebrew, add its environment variables
+    if isfile(joinpath(brew_prefix,"bin","git"))
+        ENV["GIT_EXEC_PATH"] = joinpath(brew_prefix,"opt","git","libexec","git-core")
+        ENV["GIT_TEMPLATE_DIR"] = joinpath(brew_prefix,"opt","git","share","git-core")
+    end
+    return
+end
+
+
+"""
+make_version(name::String, vers_str::String)
+
+Massage a version string from Homebrew into a valid Julia `VersionNumber` as
+best we can.  Homebrew will occasionally give us version numbers such as
+"1.0.2h_1", which doesn't map cleanly to a SemVer versioning system.  This
+function does its best to fit things into SemVer, and then tacks everything else
+onto the end as a `+` extension, e.g. the previous example maps to v"1.0.2-h+1".
+
+Also special-cases some version numbers that we know are bizarre such as the
+x264 formula, which uses dates as version numbers.
+"""
+function make_version(name::AbstractString, vers_str::AbstractString)
+    # Most of the time we fail due to weird stuff at the end; let's cut it off until it works!
+    failing = true
+    vers = nothing
+    idx = length(vers_str)
+    while failing && idx > 0
+        try
+            vers = convert(VersionNumber, vers_str[1:idx])
+            failing = false
+        catch
+            idx -= 1
+        end
+    end
+
+    if idx != 0
+        # If there's some that we chopped off, see how much we can restore via +
+        if idx < length(vers_str)
+            idx_add = idx+2
+            passing = true
+            while passing && idx_add <= length(vers_str)
+                try
+                    vers = convert(VersionNumber, "$(vers_str[1:idx])+$(vers_str[idx+2:idx_add])")
+                    idx_add += 1
+                catch
+                    passing = false
+                end
+            end
+        end
+    else
+        # Special-case things we know how to deal with here
+        if name == "x264"
+            vers = convert(VersionNumber, vers_str[2:end])
+        else
+            warn("Brew is feeding us a weird version string for $(name): $(vers_str)")
+            warn("Please report this at https://github.com/JuliaLang/Homebrew.jl")
+            vers = v"1.0"
+        end
+    end
+
+    return vers
+end
+
+
+"""
+normalize_name(name::String)
+
+Given package `name`, checks if `name` has a tap on its front and if so, taps it.
+If no tap is present, checks if the `staticfloat/juliadeps` tap should be.
+
+Returns the normalized name.
+"""
+function normalize_name(name::String)
```

Comment:
if `json` is going to take other subtypes of `AbstractString`, then this (and `tap_overrides`) probably should too